src/ConnectedMode/Migration/IMigrationPrompt.cs

```diff
@@ -78,25 +77,24 @@ public async Task ShowAsync(BoundSonarQubeProject oldBinding, bool hasNewBinding
         {
             this.oldBinding = oldBinding;
 
-            await threadHandling.RunOnUIThreadAsync(() =>
```

Comment:
The `ISolutionInfoProvider` and `INotificationService` both handle threading correctly, so this class doesn't need to do anything extra at all.

(As it happens, it will be called on the UI thread, but that's down to the caller. This class no longer cares.)